### PR TITLE
En_Insect

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_En_Insect/z_en_insect.c [hunk 1]
```diff
@@ -36,74 +44,464 @@ const ActorInit En_Insect_InitVars = {
...
+    f32 temp_f2;
+
+    Math_SmoothStepToF(&this->actor.speedXZ, 0.0f, 0.1f, 0.5f, 0.0f);
+
+    temp_f2 = (Rand_ZeroOne() * 0.8f) + (this->actor.speedXZ * 1.2f);
+    if (temp_f2 < 0.0f) {
+        this->skelAnime.playSpeed = 0.0f;
+    } else {
+        f32 clamped = CLAMP_MAX(temp_f2, 1.9f);
+        this->skelAnime.playSpeed = clamped;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Insect/z_en_insect.c:188]:
```suggestion
    } else {
        f32 clamped = CLAMP_MAX(temp_f2, 1.9f);
        
        this->skelAnime.playSpeed = clamped;
```

File: src/overlays/actors/ovl_En_Insect/z_en_insect.c [hunk 2]
```diff
@@ -36,74 +44,464 @@ const ActorInit En_Insect_InitVars = {
...
+    this->actionFunc = func_8091B07C;
+    this->unk_30C |= 0x100;
+}
+
+void func_8091B07C(EnInsect* this, GlobalContext* globalCtx) {
+    s32 pad;
+    f32 speed;
+    s16 frames;
+    s16 yaw;
+    s32 sp38 = this->actor.xzDistToPlayer < 40.0f;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Insect/z_en_insect.c:258]: Is this temp needed?

File: src/overlays/actors/ovl_En_Insect/z_en_insect.c [hunk 3]
```diff
@@ -36,74 +44,464 @@ const ActorInit En_Insect_InitVars = {
...
+void func_8091B2D8(EnInsect* this, GlobalContext* globalCtx) {
+    if ((this->unk_312 == 20) && !(this->unk_30C & 4)) {
+        this->actor.draw = EnInsect_Draw;
+    } else if (this->unk_312 == 0) {
+        if (this->unk_30C & 4) {
+            Actor_MarkForDeath(&this->actor);
+        } else {
+            Actor_SetScale(&this->actor, 0.01f);
+            func_8091AC78(this);
+        }
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Insect/z_en_insect.c:313]:
```suggestion
        if (this->unk_30C & 4) {
            Actor_MarkForDeath(&this->actor);
            return;
        }
        Actor_SetScale(&this->actor, 0.01f);
        func_8091AC78(this);
```

File: src/overlays/actors/ovl_En_Insect/z_en_insect.c [hunk 4]
```diff
@@ -36,74 +44,464 @@ const ActorInit En_Insect_InitVars = {
...
+    } else if (this->unk_312 < 20) {
+        Actor_SetScale(&this->actor, CLAMP_MAX(this->actor.scale.x + 0.001f, 0.01F));
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Insect/z_en_insect.c:315]:
```suggestion
        Actor_SetScale(&this->actor, CLAMP_MAX(this->actor.scale.x + 0.001f, 0.01f));
```

File: src/overlays/actors/ovl_En_Insect/z_en_insect.c (hunk 2 above)
Comment (Kelebek1) [reply src/overlays/actors/ovl_En_Insect/z_en_insect.c:258]: Yep.

**Review by jpburnett — APPROVED** [review]
intentional bugs in code 🐞

**Review by AngheloAlf — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**